docker/github/__init__.py

```diff
@@ -109,9 +109,11 @@ async def verify_ref_is_deployed_in_previous_environment(
         self, ref: str, environment: str, ordered_environments: Sequence[str],
     ):
         index = ordered_environments.index(environment)
-        previous_environment = ordered_environments[index - 1] if index != 0 else None
+        if index == 0:
+            return
 
-        if previous_environment is not None and await self.is_deployed_in_environment(ref, previous_environment):
+        previous_environment = ordered_environments[index - 1]
+        if not await self.is_deployed_in_environment(ref, previous_environment):
```

Comment:
I find it extremely difficult to review such fixes without any description of what the problem was and why it is fixed this way.

From the code I understand `not` in if is new, and you use early return instead of short-circuit evaluation. It's not super-clear to my, why you could't just add a `not` to the old code.

Reply:
`(index == 0) == (previous_environment is None)` was a tautology before these changes; I also found the if condition expression a bit lengthy before, so I decided to reduce complexity while fixing this function.

Reply:
Ok, I'm not a fan of early returns, but I can live with this version as well. I only wish that these changes would habe been split in two separate commits:

1. Fixing the code with `not`
2. Refactoring the code to use early return

Then it is much easier to understand for someone who didn't write the code, and I would have approved this directly, because it is clear why the first change fixes the bug, and that the second change is a cosmetic one.